src/crimson/net/SocketConnection.cc

```diff
@@ -69,6 +69,12 @@ bool SocketConnection::peer_wins() const
   return (messenger.get_myaddr() > peer_addr || policy.server);
 }
 
+seastar::future<> SocketConnection::send(MessageURef msg)
+{
+  assert(seastar::this_shard_id() == shard_id());
+  return protocol->send(MessageRef{msg.release()});
```

Comment:
I think we will finally need to switch to use `MessageURef` internally. It looks wierd to me that we need to covert an `unique_ptr` from/to an `intrusive_ptr`.

Reply:
Definitely, I agree. For this PR I wanted to first refactor the send method and its users (admin, mon, etc). The next step would be to take this a step further and refactor the internals of Messenger. Do you think it should be part of this PR?

Reply:
Either way works for me.

Reply:
Ok. Refactoring the internals will not be part of this PR but rather a follow up PR in the future.